charts/ocis/values.yaml

```
@@ -21,170 +47,397 @@ image:
 # InitContainer image used for oCIS services.
 # Only used if `services.xxx.persistence.chownInitContainer.enabled` is set to true for at least one service.
 initContainerImage:
+  # @schema
+  # type: string
+  # required: true
+  # @schema
   # -- Image repository
   repository: busybox
+  # @schema
+  # type: string
+  # required: true
+  # @schema
   # -- Image tag.
   tag: "stable"
+  # @schema
+  # type: [string, null]
+  # required: false
+  # @schema
   # -- Image sha / digest (optional).
-  sha: ""
+  sha:
+  # @schema
+  # type: [string, null]
+  # required: true
+  # @schema
```

We should somehow limit the allowed strings here, as per Kubernetes API this can only be `Always`, `Never` or  `IfNotPresent`.

```suggestion
  sha:
  # @schema
  # enum: ["Always", "Never", "IfNotPresent", null]
  # required: true
  # @schema
```

---

adding `null` to to the enum field does actually not end up in the final schema.

This would be no blocker for `.Values.image.pullPolicy`, because it is never null.

But `.Values.services.<service>.image.pullPolicy` needs to be nullable to fallback to `.Values.image.pullPolicy`

With your changes, liniting fails:

```
DEBU[2024-12-09T08:01:38+01:00] Using template files [charts/ocis/docs/templates/values-desc-table.adoc.gotmpl] for chart charts/ocis 
/home/wkloucek/go/bin/gomplate-v3.11.8 --file=charts/ocis/docs/templates/values.adoc.yaml.gotmpl  --out=charts/ocis/docs/values.adoc.yaml
/home/wkloucek/go/bin/helm-v3.16.2 lint charts/ocis -f 'charts/ocis/ci/absolute-minimum-values.yaml'
==> Linting charts/ocis
[INFO] Chart.yaml: icon is recommended
[ERROR] values.yaml: - services.activitylog.image.pullPolicy: services.activitylog.image.pullPolicy must be one of the following: "Always", "Never", "IfNotPresent"

```

As said, this is because the schema doesn't inherit the null:

```
                 "pullPolicy": {
                   "default": "",
                   "description": "Image pull policy",
+                  "enum": [
+                    "Always",
+                    "Never",
+                    "IfNotPresent"
+                  ],
                   "required": [],
-                  "title": "pullPolicy",
-                  "type": [
-                    "string",
-                    "null"
-                  ]
+                  "title": "pullPolicy"
                 },
```

---

After some further reading I guess we need to set `type` and `enum`. With `type` set to `string` and `null` as before and `enum` with the allowed string values. This should do the trick.

---

> With `type` set to `string` and `null` as before and `enum` with the allowed string values.

`FATA[2025-01-20T08:20:48+01:00] Error while validating jsonschema of key pullPolicy: cannot use both 'enum' and 'type' in the same schema` 

-> seems not to be possible.